docs/advanced/topology.md

```diff
@@ -0,0 +1,125 @@
+---
+title: Topology API's
+{{ .EditURL }}
+---
+
+The Topology feature set enables core Clutch capabilities such as autocomplete,
+as well as providing API's that can be leveraged for a multitude of purposes.
+
+One of the main goals of the topology service was to create an extensable caching mechanism,
+to store all aspects of infratructure with the ability for it to be easily accessed via API's.
+This allows implementors to write feature that are not bound to service providers API rate limits and latencies among other factors.
+Giving Clutch the control to provide a consistent user experience that you can control as your user base scales.
+
+## Topology Caching
+
+At the core of the topology service is its caching functionality which is the foundation that powers its API's.
+
+You can enable the cache like so, there are other [configuration](https://github.com/lyft/clutch/blob/c3097e5ad477952bb4bb90cc1fb5a126d7434565/api/config/service/topology/v1/topology.proto#L14-L28) options that can be tuned if necessary.
+
+```yaml title="clutch-config.yaml"
+services:
+  ...
+  # The topology services does require the postgres datastore to be configured
+  - name: clutch.service.topology
+    typed_config:
+      "@type": types.google.com/clutch.config.service.topology.v1.Config
+      // highlight-next-line
+      cache: {}
+```
+
```

Comment:
There are additional [configuration](https://github.com/lyft/clutch/blob/c3097e5ad477952bb4bb90cc1fb5a126d7434565/api/config/service/topology/v1/topology.proto#L14-L28) options that can be tuned if necessary.